src/Schema/Doctrine/DoctrineSchemaManipulator.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/**
+ * This file is part of MetaModels/core.
+ *
+ * (c) 2012-2019 The MetaModels team.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * This project is provided in good faith and hope to be usable by anyone.
+ *
+ * @package    MetaModels/core
+ * @author     Christian Schiffler <c.schiffler@cyberspectrum.de>
+ * @copyright  2012-2019 The MetaModels team.
+ * @license    https://github.com/MetaModels/core/blob/master/LICENSE LGPL-3.0-or-later
+ * @filesource
+ */
+
+declare(strict_types = 1);
+
+namespace MetaModels\Schema\Doctrine;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Schema\Comparator;
+
+/**
+ * This updates the database to be in sync with the passed schema.
+ */
+class DoctrineSchemaManipulator
+{
+    /**
+     * The doctrine connection.
+     *
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * Create a new instance.
+     *
+     * @param Connection $connection The database connection to use.
+     */
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    /**
+     * Update the database to be in sync with the passed doctrine information.
+     *
+     * @param DoctrineSchemaInformation $schemaInformation The schema information.
+     *
+     * @return void
+     */
+    public function updateDatabase(DoctrineSchemaInformation $schemaInformation): void
+    {
+        $manager = $this->connection->getSchemaManager();
+        $current = $manager->createSchema();
+        $diff    = Comparator::compareSchemas($current, $schemaInformation->getSchema());
```

Comment:
Wouldn't that delete all non MetaModels related tables? So basically alls contao tables would get dropped?

Reply:
Okay, didn't know the exact behavious of `toSaveSql`.  But it would also trigger column changes which are not updated in the install tool yet, am I right? I don't know the capabilities of doctrine here if we could limit changes to `mm_*` tables here.

Reply:
Current behaviour is to:
- Ignore table drops entirely
- Ignore column drop entirely (bad, as even our managed columns currently don't get deleted - see task in description above).
- Create/alter any column that has been mentioned by a schema provider.

Reply:
Addendum: for limiting to `mm_` tables, this is not needed nor desired, as we only manipulate tables have been defined in contrast to remove what has not been defined.

This currently bears the problem of not removing obsolete tables at all, as we do not know that it "once was defined", same for columns.

Reply:
After discussing it today I my concerns got resolved. As all non metamodels tables would get marked as dropped table and dropped tables ares executed, everything is fine here.